plib/library/Rndc.php

```diff
@@ -11,66 +11,85 @@ private function _call(Modules_SlaveDnsManager_Slave $slave, $arguments, $verbos
     {
         $slaves = null === $slave ? Modules_SlaveDnsManager_Slave::getList() : [$slave];
         foreach ($slaves as $slave) {
-            $this->_call($slave, "addzone \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$slave->getRndcView()}\"" .
+            $multiView = $this->multipleView($slave->getRndcView());
+            foreach ($multiView as $view) {
+                $this->_call($slave, "addzone \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$view}\"" .
                 " \"{ type slave; file \\\"{$domain}\\\"; masters { {$slave->getMasterPublicIp()}; }; };\"");
+            }
         }
     }
-
+    
     public function updateZone($domain, Modules_SlaveDnsManager_Slave $slave = null)
     {
         $slaves = null === $slave ? Modules_SlaveDnsManager_Slave::getList() : [$slave];
         foreach ($slaves as $slave) {
-            $result = $this->_call($slave, "refresh \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$slave->getRndcView()}\"");
-            if (false === $result) {
-                $this->addZone($domain, $slave);
+            $multiView = $this->multipleView($slave->getRndcView());
+            foreach ($multiView as $view) {
+                $result = $this->_call($slave, "refresh \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$view}\"");
+                if (false === $result) {
+                    $this->addZone($domain, $slave);
+                }
             }
         }
     }
-
+    
     public function deleteZone($domain, Modules_SlaveDnsManager_Slave $slave = null)
     {
         $slaves = null === $slave ? Modules_SlaveDnsManager_Slave::getList() : [$slave];
         foreach ($slaves as $slave) {
             $slaveStatus = $this->checkStatus($slave);
             // version: 9.9.4-RedHat-9.9.4-51.el7_4.2 (none) <id:8f9657aa>
             // version: BIND 9.10.3-P4-Ubuntu <id:ebd72b3> (none)
-            $cleanFlag = (preg_match("/version: (BIND )?(9\.10\.\d+|9\.11\.\d+|9\.12\.\d+)/", $slaveStatus)) ? "-clean" : ""; 
-
-            $this->_call($slave, "delzone $cleanFlag \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$slave->getRndcView()}\"");
+            $cleanFlag = (preg_match("/version: (BIND )?(9\.10\.\d+|9\.11\.\d+|9\.12\.\d+)/", $slaveStatus)) ? "-clean" : "";
+            
+            $multiView = $this->multipleView($slave->getRndcView());
+            foreach ($multiView as $view) {
+                $this->_call($slave, "delzone $cleanFlag \"{$domain}\" \"{$slave->getRndcClass()}\" \"{$view}\"");
+            }
         }
     }
-
+    
     public function checkStatus(Modules_SlaveDnsManager_Slave $slave)
     {
         return $this->_call($slave, "status", true);
     }
+    
+    public function multipleView($view)
+    {
+        if(strpos($view, ';')!==false){
+            $multiView = explode(';', $view);
+            if(!is_array($multiView)) throw new pm_Exception("Multi-Views Error, Views given: $view");
+        } else $multiView[] = $view;
+        
+        return $multiView;
```

Comment:
It's the same as:
`return explode(';', $view);`
Please replace.

Reply:
And could you move this to the `/plib/library/Slave.php` file as the `getRndsViews()` method?